src/main/protobuf/game.proto

```diff
@@ -2,19 +2,237 @@ syntax = "proto3";
+ * For more inforamtion see GameEvent.
+ * Depending on the action being taken, different combinations of the below fields will be specified.
+ * See mapping below:
+ * GAME_ACTION_NONE --> No Action Spec
+ * INITIAL_PLACEMENT --> vertex, edge
+ * ROLL_DICE --> No Action Spec
+ * MOVE_ROBBER_AND_STEAL --> hex, other_player
+ * BUILD_ROAD --> edge
+ * BUILD_SETTLEMENT --> vertex
+ * BUILD_CITY --> vertex
+ * BUILD_DEVELOPMENT_CARD --> No Action Spec
+ * ACTIVATE_KNIGHT --> hex, other_player
+ * ACTIVATE_ROAD_BUILDING --> edges
+ * ACTIVATE_YEAR_OF_PLENTY --> ask
+ * ACTIVATE_MONOPOLY --> ask
+ * ACTIVATE_VICTORY_POINT --> No Action Spec
+ * PROPOSE_TRADE --> ask, give
+ * PORT_TRADE --> ask, give
+ * ACCEPT_TRADE --> ask, give
+ * REJECT_TRADE --> ask, give
+ * DISCARD --> give
+ * END_TURN --> No Action Spec
+ * ACKNOWLEDGE --> No Action Spec
+ **/
+message ActionSpecification {
+  string hex = 2;
+  string vertex = 3;
+  Edge edge = 4;
+  repeated Edge edges = 5; // Road Building
+  int32 other_player_position = 7; // Trade, Steal
+  repeated Resource ask = 8; // Trade, Monopoly, Year of Plenty
+  repeated Resource give = 9; // Trade, Discard
+}
+
+message Edge {
+  string v1 = 1;
+  string v2 = 2;
+}
+
+/**
+ * Specifies the result of a TakeAction RPC call.
+ * Used as a subfield of the GameEvent response.
+ * Dependingon the GameAction, there is a different payload, as specified in the mapping below
+ * GAME_ACTION_NONE, INITIAL_PLACEMENT, ACTIVATE_VICTORY_POINT, PROPOSE_TRADE, REJECT_TRADE, END_TURN, ACKNOWLEDGE   --> No Action Result
+ * ROLL_DICE --> roll, resources_transacted
+ * MOVE_ROBBER_AND_STEAL,   ACTIVATE_KNIGHT --> HiddenCard
+ * BUILD_ROAD, BUILD_SETTLEMENT, BUILD_CITY, BUILD_DEVELOPMENT_CARD,  ACTIVATE_ROAD_BUILDING, ACTIVATE_YEAR_OF_PLENTY, ACTIVATE_MONOPOLY, PORT_TRADE, ACCEPT_TRADE,  DISCARD --> resources_transacted
+ * For more information, see GameEvent
+ **/
+message ActionResult {
+  // TODO: I think this should be a 'oneof' is there any action that might include multiple of these fields as its result?
+  int32 roll = 1;
+  HiddenCard card = 2;
+  map<int32, ResourceTransaction> resources_transacted = 3; // Only shows publically transacted resources
+}
+
+message ResourceTransaction {
+  repeated Resource gain = 1;
+  repeated Resource lose = 2;
+}
+
+/**
+ * Specifies a card that exchanged hands via a steal or a newly drawn development card.
+ * In both cases, there are only certain players that can see this card.
+ * Those player positions are specified, and the name of the card is encrypted using those clients' keys.
+ * Note: Usually resources and development cards are specified as an enum, in this case they are not.
+ * The client will have to map the string to the enum value. 
+ **/
+message HiddenCard {
+  repeated int32 viewable_by_positions = 1;
+  string encrypted_name = 2; // One of: "BRICK", "ORE", "SHEEP", "WHEAT", "WOOD", "KNIGHT", "VICTORY_POINT", "MONOPOLY", "YEAR_OF_PLENTY", "ROAD_BUILDING"
```

Comment:
I actually dont think we want to use encrypted info. Seems like a pain for a client to have to decrypt.

Reply:
And how would we make it possible for two players to decrypt a stolen cards?

Reply:
Hahaha I did a lot of thinking about it and actually came to the opposite conclusion. Ie. That it's an elegant solution. That said, one reason I incorporated this pattern now, is that we can use it in one of two ways: 

1) The way it's being used now
2) Adding a `optional_name` (instead of `encrypted_name`) meaning, the `optional_name` will only be there for players who correspond to the `viewable_by_positions`. Otherwise it will be `null`. 

In the case of 2, different players would be sent different `GameEvent` messages. They would be entirely the same except for that `optional_name` field on the `HiddenCard`. Either way, building in this pattern will put the logic in the right place upfront on the Catan server, and we can choose later how we want to implement it!

Reply:
So if two different players are sent different messages why would we need encryption? We could just send perfect and imperfect info messages to the different players